### PR TITLE
Add resource for retrieving user's CommCare Analytics roles 

### PR DESCRIPTION
**Pull request by Charl1996**
Only a backend change.

## Technical Summary
[Ticket](https://dimagi.atlassian.net/browse/SC-3474)

This PR adds a resource which returns an oauth user's commcare analytics roles by domain. It is possible to only return the CCA roles for a single domain by adding a query param filter to the resource URL. 

## Feature Flag
Related to SUPERSET_ANALYTICS, but not dependant on it.

### Safety story
Local testing

### Automated test coverage
Added some tests to cover the work

### QA Plan
[QA ticket](https://dimagi.atlassian.net/browse/QA-6982)

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/api/resources/v0_5.py [hunk 1]
```diff
@@ -903,6 +904,40 @@ class Meta(object):
...
+AnalyticsRoles.__new__.__defaults__ = ('', '')
+
+
+class CommCareAnalyticsRolesResource(CorsResourceMixin, Resource):
+    domain_name = fields.CharField(attribute='domain_name', readonly=True)
+    role_names = fields.CharField(attribute='role_names', readonly=True)
+
+    def obj_get_list(self, bundle, **kwargs):
+        request = bundle.request
+        roles_by_domain = get_commcare_analytics_roles_by_user_domains(request.couch_user)
```
Comment (Charl1996) [inline corehq/apps/api/resources/v0_5.py:917]: I can't decide whether or not to add a check here for the SUPERSET_ANALYTICS feature flag.

File: corehq/apps/api/resources/v0_5.py [hunk 2]
```diff
@@ -903,6 +904,40 @@ class Meta(object):
...
 
+AnalyticsRoles = namedtuple('AnalyticsRoles', 'domain_name role_names')
+AnalyticsRoles.__new__.__defaults__ = ('', '')
+
+
+class CommCareAnalyticsRolesResource(CorsResourceMixin, Resource):
+    domain_name = fields.CharField(attribute='domain_name', readonly=True)
+    role_names = fields.CharField(attribute='role_names', readonly=True)
+
+    def obj_get_list(self, bundle, **kwargs):
```
Comment (Charl1996) [inline corehq/apps/api/resources/v0_5.py:915]: Should we add a check here for the SUPERSET_ANALYTICS feature flag on the domains?

Comment (kaapstorm) [reply corehq/apps/api/resources/v0_5.py:915]: :thinking: I think that sounds right.

**Review by kaapstorm — COMMENTED** [review]
Looking good.

**Review by kaapstorm — APPROVED** [review]

**Comment by mkangia** [comment]
> It is possible to only return the CCA roles for a single domain by adding a query param filter to the resource URL.

I think its fine to limit to one domain at a time always.
CCA would anyway pass the domain.

Since this role is set on domain membership, its a domain bound attribute of the user and not something domain independent like the first name, so it does make sense to include domain as a requirement of the request.

**Comment by Charl1996** [comment]
@mkangia 

> its a domain bound attribute

Good point. I'll update!

**Comment by Charl1996** [comment]
Closing in favour of https://github.com/dimagi/commcare-hq/pull/34987 where I took 'n slightly different approach due to [this comment](https://github.com/dimagi/commcare-hq/pull/34954#issuecomment-2291395806).

**Pull request closed without merging**